Feature/linaria (#263)
10up/10up-toolkit: packages/toolkit/README.md

By disabling "WordPress" mode, you will need to install both react and react-dom yourself and include in your final bundle.

## <a id="linaria"></a> Linaria (CSS-in-JS)
> Support for Linaria was added in 10up-toolkit 5.1.0.

> 10up-toolkit supports [Linaria](https://linaria.dev/) without any additional configuration. The main usecase for supporting linaria is to easily compile react components authored using Linaria. This can be useful if you're building a headless site and want to share react components between your front-end and Gutenberg.

Reviewer: [nit]

```suggestion
10up-toolkit supports [Linaria](https://linaria.dev/) without any additional configuration. The main usecase for supporting linaria is to easily compile React components authored using Linaria. This can be useful if you're building a headless site and want to share React components between your front-end and Gutenberg.
```

> If you want 10up-toolkit to compile linaria first install the following packages:

Reviewer: [nit]

```suggestion
If you want 10up-toolkit to compile Linaria first install the following packages:
```

```bash
npm install --save-dev @linaria/babel-preset @linaria/webpack-loader
npm install --save @linaria/core @linaria/react
```

> Once those packages are installed, 10up-toolkit will now look for any linaria usage in JavaScript files and compile the css. The compiled css will have the name of the JS entrypoint (but with a .css extension).

Reviewer: [nit]

```suggestion

Once those packages are installed, 10up-toolkit will now look for any Linaria usage in JavaScript files and compile the CSS. The compiled CSS will have the name of the JS entry point (but with a .css extension).
```

For instance, if you have a block with the following `block.json`

```json
"editorScript": "file:./index.js",
"editorStyle": "file:./index.css"
```

> 10up-toolkit will compile any linaria code from the `index.js` entry point to `index.css`. Note that CSS imported by JS entry points always gets generated with the name of the entry point e.g: `index.js => index.css`. So make sure `editorStyle` points to the right css file which should be based on `editorScript`.

Reviewer: [nit]

```suggestion
10up-toolkit will compile any Linaria code from the `index.js` entry point to `index.css`. Note that CSS imported by JS entry points always gets generated with the name of the entry point e.g: `index.js => index.css`. So make sure `editorStyle` points to the right CSS file which should be based on `editorScript`.
```

> **IMPORTANT**: We do not currently recommend using linaria for standard WordPress builds. In a headless build the block rendering and front-end styles are handled outside of WordPress, hence why the `block.json` above only cares about editor script and style.

Reviewer: ```suggestion
**IMPORTANT**: We do not currently recommend using Linaria for standard WordPress builds. In a headless build the block rendering and front-end styles are handled outside of WordPress, hence why the `block.json` above only cares about the editor script and style.
```
